modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneAwaitDataNodesTest.java

```diff
@@ -367,20 +377,30 @@ void testRemoveZoneWhileAwaitingDataNodes() throws Exception {
 
         CompletableFuture<Set<String>> dataNodesFut1 = distributionZoneManager.topologyVersionedDataNodes(zoneId, 106);
 
+        doAnswer(invocation -> {
+            If iif = invocation.getArgument(0);
+
+            byte[] key = zoneScaleUpChangeTriggerKey(zoneId).bytes();
+
+            if (Arrays.stream(iif.cond().keys()).anyMatch(k -> Arrays.equals(key, k))) {
+                //Drop the zone while dataNodesFut1 is awaiting a data nodes update.
+                distributionZoneManager.dropZone(zoneName).get();
```

Comment:
It's better to bound future awaiting time either with `.get(3, SECONDS);` or with
```
        // Drop zone.
        assertThat(distributionZoneManager.dropZone(zoneName), willSucceedIn(3, SECONDS));
```
From my point of view second approach is generally better.

Reply:
Fixed